apps/teams/forms.py

```diff
@@ -96,16 +96,21 @@ def save(self, request):
 class TeamChangeForm(forms.ModelForm):
     class Meta:
         model = Team
-        fields = ("name", "slug")
+        fields = ("name",)
         labels = {
             "name": _("Team Name"),
-            "slug": _("Team ID"),
         }
         help_texts = {
             "name": _("Your team name."),
-            "slug": _("A unique ID for your team. No spaces are allowed!"),
         }
 
+    def save(self, commit=True):
+        instance = super().save(commit=False)
+        instance.slug = get_next_unique_team_slug(instance.name)
```

Comment:
I think this should only happen if the instance is new. I don't think the slug should ever change once it's been set.

Reply:
https://github.com/dimagi/open-chat-studio/pull/487